test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/util/OptionalTest.scala

```diff
@@ -0,0 +1,70 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib.util
+
+import org.junit.Assert._
+import org.junit.Test
+
+import java.util.Optional
+
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class OptionalTest {
+
+  @Test def should_instantiate_properly_optionals(): Unit = {
+    Optional.empty[String]()
+    Optional.of[String]("")
+    assertThrows(classOf[NullPointerException], Optional.of[String](null))
+    Optional.ofNullable[String]("")
+    Optional.ofNullable[String](null)
+  }
+
+  @Test def should_properly_compare_optionals(): Unit = {
+    assertEquals(Optional.empty[String](), Optional.ofNullable[String](null))
+    assertEquals(Optional.of[String](""), Optional.ofNullable[String](""))
+    assertEquals(Optional.of[Int](1), Optional.ofNullable[Int](1))
+    case class Test(value: Long)
+    assertEquals(Optional.of(Test(1L)), Optional.ofNullable(Test(1L)))
+  }
```

Comment:
There should also be some `assertNotEquals`s here, to test that `equals` returns `false` when it is supposed to return `false`.